recipe/build.sh

```diff
@@ -1,29 +1,72 @@
-#!/bin/bash
+#!/usr/bin/env bash
+set -eu
+
+# Paths are hardcoded in binaries, simplify to basename
 export CC=$(basename "$CC")
 export ASPP="$CC -c"
 export AS=$(basename "$AS")
 export AR=$(basename "$AR")
 export RANLIB=$(basename "$RANLIB")
-export OCAML_PREFIX=$PREFIX
-export OCAMLLIB=$PREFIX/lib/ocaml
-
-if [ "$(uname)" = "Darwin" ]; then
-# Tests failing on macOS. Seems to be a known issue.
-  rm testsuite/tests/lib-threads/beat.ml
-fi 
 
-bash -x ./configure -prefix $OCAML_PREFIX
-make world.opt -j${CPU_COUNT}
-make ocamltest -j ${CPU_COUNT}
+# Avoids an annoying 'directory not found'
 mkdir -p ${PREFIX}/lib
-# Check if cross-compiling - not testing on build architecture
-if [[ -z ${CONDA_BUILD_CROSS_COMPILATION} ]]; then
-  make tests
+
+if [[ "${target_platform}" != "linux-"* ]] && [[ "${target_platform}" != "osx-"* ]]; then
```

Comment:
I'd have written this as "if Windows", but not sure off hand how better to express this.